src/Compilers/CSharp/Portable/BoundTree/UnboundLambda.cs

```diff
@@ -132,10 +146,25 @@ private static TypeSymbol InferReturnType(BoundBlock block, Binder binder, bool
 
             // Async:
 
+            NamedTypeSymbol taskType = null;
```

Comment:
Could you comment what this chunk of code is doing? It's not obvious. I think it says:
// This lambda has an "async" modifier so we need to figure out what tasklike the lambda returns.
// The default assumption is it will return Task or Task<T> (depending on whether it has returns or not).
// But if the lambda is being matched against a delegate type (i.e. not a generic type parameter, and
// not a non-delegate type), and if the delegate type has a return type, and if that return type is
// tasklike (as identified by the presence of a createBuilderMethod), then the delegate's return type
// indicates which tasklike the lambda should return. It might be Task, or Task<T>, or Task<concrete>,
// or Tasklike, or Tasklike<T>, or Tasklike<concrete>.